### PR TITLE
add card api

### PR DESCRIPTION
**Pull request by chikamura**
- add card api
- add webmock

### CODE REVIEW

File: lib/embulk/input/twitter_ads_analytics.rb [hunk 1]
```diff
@@ -43,23 +44,25 @@ class ServiceUnavailable < ServerError; end
...
+          "granularity" => config.param("granularity", :string, **optional_if_card)&.upcase,
+          "placement" => config.param("placement", :string, **optional_if_card)&.upcase,
+          "start_date" => config.param("start_date", :string, **optional_if_card),
+          "end_date" => config.param("end_date", :string, **optional_if_card),
+          "timezone" => config.param("timezone", :string, **optional_if_card),
           "entity_start_date" => config.param("entity_start_date", :string, default: nil),
           "entity_end_date" => config.param("entity_end_date", :string, default: nil),
           "entity_timezone" => config.param("entity_timezone", :string, default: nil),
-          "async" => config.param("timezone", :bool),
+          "async" => config.param("timezone", :bool, **optional_if_card),
```
Comment (d-hrs) [inline lib/embulk/input/twitter_ads_analytics.rb]: must-badge

```suggestion
          "async" => config.param("async", :bool, **optional_if_card),
```

File: lib/embulk/input/twitter_ads/api.rb [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+              retry
+            else
+              logger.error("exceeds the upper limit retry, #{e.message}")
+              raise e
+            end
+          end
+        end
+
+        private
+        def get_sleep_sec(response:, retries:)
```
Comment (d-hrs) [inline lib/embulk/input/twitter_ads/api.rb:62]: memo-badge

duplicated with the following code:
https://github.com/trocco-io/embulk-input-twitter_ads_analytics/blob/44dd25302f46746f4fa670322b67a0b0e4bd52d5/lib/embulk/input/twitter_ads_analytics.rb#L408-L422

File: lib/embulk/input/twitter_ads/card.rb [hunk 3]
```diff
@@ -0,0 +1,63 @@
+require 'json'
+require_relative 'api'
+require_relative 'util'
+
+module Embulk
+  module Input
+    module TwitterAds
+      class Card
```
Comment (d-hrs) [inline lib/embulk/input/twitter_ads/card.rb:8]: good-badge

👍 
we need refactoring to create a class for each resource.

**Review by d-hrs — CHANGES_REQUESTED** [review]
fix async option please.

**Review by u110 — APPROVED** [review]

**Review by d-hrs — APPROVED** [review]
LGTM

**Pull request merged**